src/batchUploader.ts

```diff
@@ -136,7 +136,11 @@ export class BatchUploader {
 
                 if (onCreateBatchCallback) {
                     uploadBatchObject = onCreateBatchCallback(uploadBatchObject);
-                    uploadBatchObject.mutated = true;
+                    if (uploadBatchObject) {
```

Comment:
Would it make useful to check if the `uploadBatchObject` is actually mutated rather than if it only exists?